src/bitmap/store/array_store/vector.rs

```diff
@@ -0,0 +1,766 @@
+//! Ported from CRoaring and arXiv:1709.07821
+//! Lemire et al, Roaring Bitmaps: Implementation of an Optimized Software Library
+//!
+//! Prior work: Schlegel et al., Fast Sorted-Set Intersection using SIMD Instructions
+//!
+//! Rust port notes:
+//! The x86 PCMPESTRM instruction been replaced with a simple vector or-shift
+//! While several more instructions, this is what is available through LLVM intrinsics
+//! and is portable.
+
+#![cfg(feature = "simd")]
+
+use super::scalar;
+use simd::{mask16x8, simd_swizzle, u16x8, LaneCount, Mask, Simd, SimdElement, SupportedLaneCount};
+
+// a one-pass SSE union algorithm
+pub fn or(lhs: &[u16], rhs: &[u16], visitor: &mut impl BinaryOperationVisitor) {
+    // De-duplicates `slice` in place
+    // Returns the end index of the deduplicated slice.
+    // elements after the return value are not guaranteed to be unique or in order
+    #[inline]
+    fn dedup(slice: &mut [u16]) -> usize {
+        let mut pos: usize = 1;
+        for i in 1..slice.len() {
+            if slice[i] != slice[i - 1] {
+                slice[pos] = slice[i];
+                pos += 1;
+            }
+        }
+        pos
+    }
```

Comment:
Yeah, though in this specific instance case we can backshift as we do here, instead of swapping to the back.  We don't care about the back half of the binary partition.

Reply:
Besides! This whole module is nightly only due to core simd.